spec/abci++/abci++_properties_001_draft.md

```diff
@@ -0,0 +1,675 @@
+    | byzantine_validators    | repeated [Evidence](#evidence)              | List of evidence of validators that acted maliciously.                                                     | 4            |
+    | tx                      | repeated bytes                              | Preliminary list of transactions that have been picked as part of the block to propose.                    | 5            |
+
+>**TODO**: DISCUSS: We need to make clear whether a proposer is also running the logic of a non-proposer node (in particular "ProcessProposal")
+From the App's perspective, they'll probably skip ProcessProposal
+
+* **Response**:
+
+    | Name                    | Type                                             | Description                                                                                 | Field Number |
+    |-------------------------|--------------------------------------------------|---------------------------------------------------------------------------------------------|--------------|
+    | modified                | bool                                             | The Application sets it to true to denote it did not make changes                           | 1            |
+    | tx                      | repeated [TransactionRecord](#transactionrecord) | Possibly modified list of transactions that have been picked as part of the proposed block. | 2            |
+    | data                    | bytes                                            | The Merkle root hash of the application state.                                              | 3            |
+    | validator_updates       | repeated [ValidatorUpdate](#validatorupdate)     | Changes to validator set (set voting power to 0 to remove).                                 | 4            |
+    | consensus_param_updates | [ConsensusParams](#consensusparams)              | Changes to consensus-critical gas, size, and other parameters.                              | 5            |
+
+* **Usage**:
+    * The Application can modify the parameters received in `RequestPrepareProposal` before sending
+    them in `ResponsePrepareProposal`. In that case, `ResponsePrepareProposal.modified` is set to true.
+    * In same-block execution mode, the Application can (and should) modify `ResponsePrepareProposal.data`,
+      `ResponsePrepareProposal.validator_updates`, and `ResponsePrepareProposal.consensus_param_updates`.
+    * In next-block execution mode, the Application can only modify `ResponsePrepareProposal.tx`, Tendermint
+      will ignore any modification to the other fields.
+    * If `ResponsePrepareProposal.modified` is false, then Tendermint should ignore the rest of
+      parameters in `ResponsePrepareProposal`.
+    * As a sanity check, Tendermint will check the returned parameters for validity if the Application modified them.
```

Comment:
These seem to capture the two main possibilities. My preference is for the second option, since we don't really have a consensus failure if a proposer behaves incorrectly, but I think we should get others to way in on this as well.

Reply:
Ok, I'll add this as a TODO in the text so other reviewers can voice their opinion